fulp_modules/features/toys/code/shrimp_plush.dm

```diff
@@ -0,0 +1,37 @@
+/datum/mood_event/shrimp_rice
+	description = "I can't believe a shrimp fried this rice!"
+	mood_change = 3
+	timeout = 2 MINUTES
+
+/obj/item/toy/plush/shrimp/Destroy()
+	. = ..()
+	if(fried_rice)
+		UnregisterSignal(fried_rice, COMSIG_FOOD_EATEN)
+
+/obj/item/toy/plush/shrimp/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(proximity_flag && has_fried)
+		return
+
+	if(!istype(user.mind.assigned_role, /datum/job/cook))
+		return
+
+	if(!istype(target, /obj/item/food))
+		return
+
+	var/obj/item/food/target_food = target
+	if(!findtext(target.name, "fried rice"))
+		return
+
+	RegisterSignal(target_food, COMSIG_FOOD_EATEN, PROC_REF(on_eaten))
+	has_fried = TRUE
+	fried_rice = target_food
+	balloon_alert(user, "fried rice!")
```

Comment:
This is all weird, why doesn't the rice just have a boolean var ``fried_by_shrimp`` or something to give the mood on being eaten, there's no need to hold reference to anything like this.

Reply:
Didn't want to touch tg code, and there's no actual subtype of food that is fried rice